File: tests/CoreTests/Methods/NamedParameters.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using SharpKit.JavaScript;

namespace CoreTests.Methods
{

    [JsType(JsMode.Prototype)]
    class NamedParameters
    {
        public void Foo(string prm1, string prm2, string prm3)
        {
        }

        void Foo2()
        {
            Foo(prm3 : "prm3", prm2 : "prm2", prm1 : "prm1");
        }
    }
}

File: trunk/tests/CoreTests/Methods/NamedParameters.js

if (typeof(CoreTests) == "undefined")
    var CoreTests = {};
if (typeof(CoreTests.Methods) == "undefined")
    CoreTests.Methods = {};
CoreTests.Methods.NamedParameters = function (){
};
CoreTests.Methods.NamedParameters.prototype.Foo = function (prm1, prm2, prm3){
};
CoreTests.Methods.NamedParameters.prototype.Foo2 = function (){
    this.Foo("prm1", "prm2", "prm3");
};

